client/src/pages/EditProfile/EditProfile.tsx

```diff
@@ -0,0 +1,236 @@
+import React, { useState } from 'react';
+import {
+  Button,
+  CssBaseline,
+  TextField,
+  Grid,
+  FormControl,
+  MenuItem,
+  Select,
+  Typography,
+  Container,
+  Paper,
+  Box,
+} from '@material-ui/core';
+import useStyles from './useStyles';
+import { FormikHelpers, useFormik } from 'formik';
+import * as yup from 'yup';
+import profile from '../../helpers/APICalls/profile';
+import { IProfile } from '../../helpers/APICalls/profile';
+import { useSnackBar } from '../../context/useSnackbarContext';
+
+const validationSchema = yup.object({
+  firstName: yup.string().required('First Name is required'),
+  lastName: yup.string().required('Last Name is required'),
+  email: yup.string().email('Enter a valid email').required('Email is required'),
+  phone: yup.string(),
+  address: yup.string().required('Address is required'),
+  description: yup.string().required('Your description is required'),
+});
+
+export default function EditProfile(): JSX.Element {
```

Comment:
If a user has an existing profile maybe we can fetch it within this component? In case he does have a profile, we can update it. However, if he does not have a profile, we can create one for him :)